### PR TITLE
feat(youtubeIntegration): #22 Implemented functionalities to integrate youtube

### PR DESCRIPTION
**Pull request by VenkatVenkatesh**
- Implemented functionalities to login via youtube.

- Added functionalities to connect youtube as a link Account.
Screen Shot 2019-05-10 at 10 55 01 AM

### CODE REVIEW

File: web/src/app/components/social-cards/social-cards.component.ts [hunk 1]
```diff
@@ -52,6 +54,14 @@ export class SocialCardsComponent implements OnInit {
       });
   }
 
+  addYoutubeAccount(): void {
+    this.accountsService.linkWithYoutube()
+      .subscribe((response) => {
+        this.saveSecondaryUser(response, 'youtube');
```
Comment (eddiejaoude) [inline web/src/app/components/social-cards/social-cards.component.ts]: This can go on a single line

File: web/src/app/models/youtube.model.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface Youtube {
+  etag: string;
+  items: Array<string>;
```
Comment (eddiejaoude) [inline web/src/app/models/youtube.model.ts]: I think this way of defining an array has been deprecated, and should now be `string[]`, but not a big problem.

File: web/src/app/services/auth.service.ts [hunk 3]
```diff
@@ -66,11 +96,51 @@ export class AuthService {
...
+      followers: response.additionalUserInfo.profile.followers ? response.additionalUserInfo.profile.followers :
+        response.pageInfo.totalResults,
+      following: response.additionalUserInfo.profile.following ? response.additionalUserInfo.profile.following : 0,
+      avatarUrl: response.additionalUserInfo.profile.avatar_url ? response.additionalUserInfo.profile.avatar_url :
+        response.additionalUserInfo.profile.picture
+    };
+
+    const additionalUserInfo = {
+      username: response.additionalUserInfo.profile.name,
+      profile
```
Comment (eddiejaoude) [inline web/src/app/services/auth.service.ts]: Missing trailing comma

File: web/src/app/app.module.ts [hunk 4]
```diff
@@ -48,6 +48,7 @@ import { SocialCardsComponent } from './components/social-cards/social-cards.com
     UserService,
     ErrorService,
     AuthGuardService,
+    HttpClientModule
```
Comment (eddiejaoude) [inline web/src/app/app.module.ts]: I don't think a module is needed here?

File: web/src/app/components/social-cards/social-cards.component.ts (hunk 1 above)
Comment (VenkatVenkatesh) [reply web/src/app/components/social-cards/social-cards.component.ts]: Okay

File: web/src/app/app.module.ts (hunk 4 above)
Comment (VenkatVenkatesh) [reply web/src/app/app.module.ts]: Okay 👍

Comment (eddiejaoude) [reply web/src/app/app.module.ts]: I think it needs to be under the `imports` but not the `providers`. But if it still works then it is fine I guess

File: web/src/app/app.module.ts [hunk 5]
```diff
@@ -32,6 +32,7 @@ import { SocialCardsComponent } from './components/social-cards/social-cards.com
     HomeComponent,
     NavbarComponent,
     SocialCardsComponent,
+    HttpClientModule
```
Comment (eddiejaoude) [inline web/src/app/app.module.ts]: `declarations` I think is incorrect and it should be under `imports`. See our other discussion https://github.com/DashboardHub/TribeDashboard/pull/76#discussion_r283222655

Comment (VenkatVenkatesh) [reply web/src/app/app.module.ts]: Apologies @eddiejaoude. Changed it.

**Review by eddiejaoude — APPROVED** [review]

**Pull request merged**